### ISSUE TITLE
Doesn't work with docker machine

### ISSUE THREAD

**Issue by srid** [jwilder/nginx-proxy#88]
I created a [docker 'machine'](https://github.com/docker/machine) on Digital Ocean. nginx-proxy is unable to access the host docker daemon. 

This is how I ran nginx-proxy:

```
docker run -d -p 80:80 -v /var/run/docker.sock:/tmp/docker.sock jwilder/nginx-proxy
```

This is what I see in the logs:

```
dockergen.1 | 2015/01/16 01:57:17 Unable to ping docker daemaon: dial unix /tmp/docker.sock: connection refused
dockergen.1 | 2015/01/16 01:57:27 Unable to ping docker daemaon: dial unix /tmp/docker.sock: connection refused
dockergen.1 | 2015/01/16 01:57:37 Unable to ping docker daemaon: dial unix /tmp/docker.sock: connection refused
```

**Comment by srid** [comment]
This is what I see on the server-side docker setup (the 'machine'):

```
# grep -v ^\# /etc/default/docker | grep ^e
export DOCKER_OPTS="--auth=identity --host=tcp://0.0.0.0:2376"
```

The file `/var/run/docker.sock` exists though. But it cannot be used with the local docker client:

```
# docker -H unix:///var/run/docker.sock ps
FATA[0000] Cannot connect to the Docker daemon. Is 'docker -d' running on this host? 
#
```

(I should note that I'm using the beta build of docker client linked from the README of the docker machine repo)

**Comment by md5** [comment]
@srid: I think the machine folks fixed that here: https://github.com/docker/machine/pull/259

You might be able to work around by setting `-e DOCKER_HOST=tcp://0.0.0.0:2376` on your `nginx-proxy` container, but it would probably be better to get a more up-to-date build of machine and create a new DO machine and/or update the `DOCKER_OPTS` to include `--host=unix:///var/run/docker.sock` in your `/etc/default/docker`. Going the `DOCKER_HOST` route would also mean trying to make the `docker-gen` instance in the `nginx-proxy` container speak TLS to the Docker service, which sounds like a pain in the ass to me.

**Comment by md5** [comment]
Regarding my comment about TLS, you'd actually need to get `docker-gen` to authenticate via `identity` auth since machine isn't using TLS by default at the moment (that's in https://github.com/docker/machine/pull/255).

**Comment by ehazlett** [comment]
@md5 Yes are moving (for now) to the standard TLS mechanism that is currently supported in Docker as you referenced above.  If you want to have Identity support in the future, I've added some helper functions to docker/libtrust (https://github.com/docker/libtrust/commit/1dc3c57f6be6c80ea80441e6fb11b320f6da8a6a) that you can use with https://github.com/samalba/dockerclient out of the box -- no modifications needed in dockerclient.  You simply create the TLSConfig from the Identity trust keys and pass that to dockerclient and it's good.

Also, thanks for trying Machine :)

**Comment by md5** [comment]
Thanks @ehazlett

**Comment by JamBalaya56562** [comment]
Going through some long-open issues as a third party. This one looks like it has been overtaken by events. The root cause was in Docker Machine (the daemon only listening on `tcp://0.0.0.0:2376` with `--auth=identity` rather than on the Unix socket), and the thread itself identified it as a Machine-side problem with an upstream fix (docker/machine#259) and a `DOCKER_HOST` workaround. Docker Machine has since been deprecated/archived, and the `identity`-auth blocker mentioned here was already being replaced by standard TLS at the time.

On current nginx-proxy (v1.11.2), connecting to a remote/TLS daemon is a first-class, documented option via `DOCKER_HOST`, `DOCKER_TLS_VERIFY` and `DOCKER_CERT_PATH` (see the docker-gen environment variables in the docs), which covers exactly the `DOCKER_HOST=tcp://...` approach suggested above. There is also the separate-containers / docker-socket-proxy pattern for not exposing the socket directly.

Given that, this may be a candidate for closing if a maintainer agrees, though I'll leave that judgement to them. If anyone is still hitting this with a current setup, a fresh reproduction would be helpful.
